universe/ShipDesign.cpp

```diff
@@ -675,9 +675,10 @@ void ShipDesign::BuildStatCaches() {
         { m_num_part_classes[part_class]++; }
     }
 
-    // ensure tags are unique and copy into this->m_tags
+    // ensure tags are unique
     std::sort(tags.begin(), tags.end());
     auto last = std::unique(tags.begin(), tags.end());
+    tags.erase(last, tags.end());
```

Comment:
The erase is unnecesary, or at least not the best solution, I think. `tags` is a `vector<string_view>` assembled from the tags of the hull and parts. Some of those could be duplicates. `sort` and `unique` arrange all the unique ones at the start. Later, they are copied into `m_tags_concatenated`, but instead of erasing, it can just iterate from `begin` to `last` since that should be one-past the last unique tag in the vector.

Reply:
Ah, got it! That makes sense now.